pennylane/qnode.py

```diff
@@ -890,14 +916,18 @@ def __call__(self, *args, **kwargs) -> qml.typing.Result:
 
                 # pylint: disable=not-callable
                 # update the gradient function
-                set_shots(self._original_device, override_shots)(self._update_gradient_fn)()
+                if isinstance(self._original_device, Device):
+                    set_shots(self._original_device, override_shots)(self._update_gradient_fn)()
 

```

Comment:
Maybe add a comment that we would support this way of passing shots? (total number) I am not sure we want that and also there was some issue with jitting

Reply:
I just updated it so that `_update_gradient_fn` can now take the shots as a keyword instead of from the device.

If there's a failure, mind sending it on?

Reply:
I think https://github.com/PennyLaneAI/pennylane/issues/3474

Reply:
Thanks.

Looks like there's a couple of things we will need to do to allow jitting:

1) Make it so that shots can accept an abstract tracer
2) Improve the `shape` methods to work with the new location for shots
3) Potentially more.

But this is a first step.